### DISCUSSION TITLE
Nats jetstream

### DISCUSSION THREAD

**Discussion by bgervan** [celery/kombu#2100] [Ideas]
Hey,

Is there any plan to support nats jetstream as transport protocol?

**Comment by TayfurYldz** [comment]
There isn't a NATS/JetStream transport in Kombu at the moment. The current transport registry has no NATS entry, so this would be a new transport rather than a broker URL/configuration that can be enabled today.

JetStream has the primitives needed for a queue-like mapping (durable consumers, explicit acknowledgements, redelivery and acknowledgement deadlines), but the important part would be defining Kombu semantics rather than merely publishing/subscribing: queue/exchange routing, QoS/prefetch, ack/reject/requeue, visibility/redelivery behavior, and what features the transport advertises.

So for anyone pursuing this, I think the useful first artifact would be a transport design mapping those Kombu operations to JetStream consumer/stream operations, followed by the transport conformance tests used by the other virtual transports. A thin Core NATS pub/sub adapter would not provide the delivery guarantees Celery expects.
